pagopa/io-app
# chore(Cross): [IOAPPX-210] Upgrade `react-navigation` to v6

## Short description
This PR upgrades react-navigation to `6.x` with all its peer dependencies

**refs:** 
- https://reactnavigation.org/docs/upgrading-from-5.x/
- https://reactnavigation.org/docs/troubleshooting/

## List of changes proposed in this pull request
- Version upgrade
- Components refactoring to handle the deprecated navigation prop

## How to test
```
rm -rf node_modules
cd ios && bundle exec pod deintegrate && cd .. 
yarn
cd ios && bundler exec pod install && cd .. 
yarn react-native start --reset-cache
yarn reat-native run-yourplatform
```
Run tests and check all of the main app flows
